bin/node/src/flags/p2p.rs

```diff
@@ -304,11 +304,7 @@ impl P2PArgs {
         let advertise_ip = self.advertise_ip.unwrap_or(self.listen_ip);
 
         // If the advertise tcp port is null, use the listen tcp port
-        let advertise_tcp_port = if self.advertise_tcp_port != 0 {
-            self.advertise_tcp_port
-        } else {
-            self.listen_tcp_port
-        };
+        let advertise_tcp_port = self.advertise_tcp_port.unwrap_or(self.listen_tcp_port);
```

Comment:
I don't think this is going to work as intended. Since the `default_value = 0`, from the CLI, you won't be able to disable the advertise port. Let's add unit tests here to verify e.g. `mock-command --p2p.advertise.tcp none` and without the flag specified (default).